distbench_utils.cc

```diff
@@ -82,8 +82,15 @@ void InitLibs(const char* argv0) {
   GOOGLE_PROTOBUF_VERIFY_VERSION;
 }
 
-absl::StatusOr<DeviceIpAddress> IpAddressForDevice(std::string_view netdev) {
-  return GetBestAddress(use_ipv4_first, netdev);
+absl::StatusOr<DeviceIpAddress> IpAddressForDevice(std::string_view netdev,
+                                                   int ip_version) {
+  if (ip_version == 4) {
+    return GetBestAddress(true, netdev);
+  } else if (ip_version == 6) {
+    return GetBestAddress(false, netdev);
```

Comment:
For the case we request a specific version, shouldn't we check that we only get the version requested and fail if not ?

Reply:
Yes. We just fail in the caller (homa) for now.

Reply:
fixing this now, will add one more commit before pushing.